test/integration/targets/expect/tasks/main.yml

```diff
@@ -241,3 +241,106 @@
     that:
     - "non_zero_result.failed"
     - "non_zero_result.msg == 'non-zero return code'"
+
+- name: run tests in venv
+  vars:
+    venv_dir:  "{{ output_dir }}/pexpected_venv"
+    venv_python: "{{ venv_dir }}/bin/python"
+  block:
+    - name: ensure venv exists
+      pip:
+        name: coverage
+        virtualenv: "{{ venv_dir }}"
+        virtualenv_command: "{{ ansible_python_interpreter }} -m venv"
+
+    - name: trigger pexpect dependency absence
+      expect:
+        command: "echo expectThePexpected"
+        responses:
+          foo: bar
+      vars:
+        ansible_python_interpreter: "{{ venv_python }}"
+      register: pexpect_dependency_absence
+      ignore_errors: true
+
+    - name: assert pexpect dependency absence
+      assert:
+        that:
+          - "pexpect_dependency_absence.failed"
+          - "'pexpect' in pexpect_dependency_absence.msg"
+
+    - name: Install pexpect3.3
+      pip:
+        break_system_packages: true
+        name: pexpect
+        version: 3.3
+        virtualenv: "{{ venv_dir }}"
+
+    - name: trigger fallback to pexpect3.3 methods
+      expect:
+        command: "echo pexpectIsABitOutOfDate"
+        responses:
+          foo: bar
+      vars:
+        ansible_python_interpreter: "{{ venv_python }}"
+      register: pexpect_version_three_dot_three
+
+    - name: uninstall pexpect3.3
+      pip:
+        break_system_packages: true
```

Comment:
IIRC the default pip module behavior is idempotent exact version match, so the explicit `absent` tasks between version changes should be unnecessary- it should just "make it so" for whatever version you specified.